yellowbrick/features/base.py

```diff
@@ -309,3 +312,130 @@ def _determine_target_color_type(self, y):
                 "could not determine target color type "
                 "from target='{}' to '{}'"
             ).format(self.target_type, self._target_color_type))
+
+
+##########################################################################
+## Projection Visualizers
+##########################################################################
+
+class ProjectionVisualizer(DataVisualizer):
+    
+    def __init__(self, transformer, ax=None, features=None, classes=None, color=None,
+             colormap=None, target_type="auto", proj_dim=2, alpha=0.75, **kwargs):
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        self.transformer=transformer
+        if proj_dim not in [2, 3]:
+            raise YellowbrickValueError("Projection dimensions must be either 2 or 3")
+        self.proj_dim = proj_dim
+        self.alpha = alpha
+        self._cax = None
+
+    @property
+    def cax(self):
+        """
+        The axes of the colorbar, right of the scatterplot.
+        """
+        if self._cax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the X axis"
+            )
+        return self._cax       
+        
+    def _layout_colorbar(self):
+        """
+        Creates the layout for colorbar. The colorbar is added to the right of the 
+        scatterplot 
+        """
+        # Ensure matplotlib version compatibility
+        if make_axes_locatable is None:
+            raise YellowbrickValueError((
+                "heatmap requires matplotlib 2.0.2 or greater "
+                "please upgrade matplotlib or set heatmap=False on the visualizer"
+            ))
+
+        # Create the new axes for the colorbar
+        divider = make_axes_locatable(self.ax)
+        self._cax = divider.append_axes("right", size="5%", pad=0.3)
+        
+    def fit(self, X, y=None, **kwargs):
+        
+        self.fit_transform(X, y, **kwargs)
+        return self
+        
+    def fit_transform(self, X, y=None, **kwargs):
+
+        # Transformer will be  defined either in the subclasses or explicitly
+        self.transformed_features_ = self.transformer.fit_transform(X)
```

Comment:
We don't want to save data on the visualizer wherever we can avoid it - imagine that this is a really large text corpus, we could easily blow up the memory of a users machine if we save data. Instead, let's just pass the transformed data to `draw()`. 

Also the `super` and the `fit()` should come before transform here.

Reply:
The super `fit` calls draw(). So if I place the super `fit` before `transform` then It will call draw without having transformed data.
This is also one of the reasons I saved the data on the visualizer. The problem that arises here is that call to `draw()` is made by the `super` class. So, either we somehow pass the data to super class or save it on visualizer. Is there any other method, that you think will help us?

Reply:
Ok, so an estimator must be `fit()` before it can either `transform()` or `predict()` -- this is a pretty critical part of the sklearn workflow. It looks like we might have to remove the call to draw in the `DataVisualizer` -- this was just there as a convenience anyway. Would you mind looking and seeing what this will affect -- e.g. what subclasses of `DataVisualizer` rely on this call to `draw`? It should be pretty straight forward to fix:

```python
class MyVisualizer(DataVisualizer):

    def fit(self, X, y=None):
        super(MyVisualizer, self).fit(X, y)
        self.draw(X, y)
        return self
```

Reply:
You know, this might be why `Manifold` didn't extend `DataVisualizer` ... so this will be a good thing to fix!